tests/iamclient/iamclient_test.cpp

```diff
@@ -7,362 +7,740 @@
 
-                return grpc::Status::OK;
-            }));
+        mStream->Write(request);
+    }
 
-    EXPECT_CALL(mIAMClientMock, CreateIAMProvisioningServiceStub)
-        .WillOnce(Return(std::move(mIAMProvisioningServiceStub)));
+    void ResumeNodeRequest(const std::string& id)
+    {
+        iamanager::v5::IAMIncomingMessages request;
 
-    aos::StaticArray<aos::StaticString<aos::iam::certhandler::cCertTypeLen>, 1> certTypes;
+        request.mutable_resume_node_request()->set_node_id(id);
 
-    const auto err = mIAMClientMock.GetCertTypes(cNodeId, certTypes);
-    ASSERT_TRUE(err.Is(aos::ErrorEnum::eNoMemory)) << err.Message();
-    ASSERT_NE(certTypes.Size(), cTypes.size());
-}
+        mStream->Write(request);
+    }
+
+    void CreateKeyRequest(
+        const std::string& id, const std::string& subject, const std::string& type, const std::string& password)
+    {
+        iamanager::v5::IAMIncomingMessages request;
+
+        request.mutable_create_key_request()->set_node_id(id);
+        request.mutable_create_key_request()->set_subject(subject);
+        request.mutable_create_key_request()->set_type(type);
+        request.mutable_create_key_request()->set_password(password);
+
+        mStream->Write(request);
+    }
+
+    void ApplyCertRequest(const std::string& id, const std::string& type, const std::string& cert)
+    {
+        iamanager::v5::IAMIncomingMessages request;
+
+        request.mutable_apply_cert_request()->set_node_id(id);
+        request.mutable_apply_cert_request()->set_type(type);
+        request.mutable_apply_cert_request()->set_cert(cert);
+
+        mStream->Write(request);
+    }
+
+    MOCK_METHOD(void, OnNodeInfo, (const iamanager::v5::NodeInfo& nodeInfo));
+    MOCK_METHOD(void, OnStartProvisioningResponse, (const ::common::v1::ErrorInfo& errorInfo));
+    MOCK_METHOD(void, OnFinishProvisioningResponse, (const ::common::v1::ErrorInfo& errorInfo));
+    MOCK_METHOD(void, OnDeprovisionResponse, (const ::common::v1::ErrorInfo& errorInfo));
+    MOCK_METHOD(void, OnPauseNodeResponse, (const ::common::v1::ErrorInfo& errorInfo));
+    MOCK_METHOD(void, OnResumeNodeResponse, (const ::common::v1::ErrorInfo& errorInfo));
+    MOCK_METHOD(void, OnCreateKeyResponse,
+        (const std::string& type, const std::string& csr, const ::common::v1::ErrorInfo& errorInfo));
+    MOCK_METHOD(void, OnApplyCertResponse,
+        (const std::string& type, const std::string& certURL, const std::string& serial,
+            const ::common::v1::ErrorInfo& errorInfo));
+
+    void WaitNodeInfo()
+    {
+        std::unique_lock guard {mMutex};
+
+        mNodeInfoLock.wait(guard);
+    }
+
+    void WaitNodeInfoFor(const std::chrono::seconds& seconds)
+    {
+        std::unique_lock guard {mMutex};
+
+        mNodeInfoLock.wait_for(guard, seconds);
+    }
+
+    void WaitResponse()
+    {
+        std::unique_lock guard {mMutex};
+
+        mResponseLock.wait(guard);
```

Comment:
maybe add timeout on wait, because we can stuck on CI if message not received

Reply:
added timeout